tspp.java

```diff
@@ -55,23 +55,23 @@ void display()
 
 void dfs_1()
 {
-Stack <Integer> s=new Stack<Integer> ();
+@LTLengthOf("mat") Stack <Integer> s=new Stack<Integer> ();
 System.out.println("Enter the starting vertex: ");
 
-@LTLengthOf("visited") int start=sc.nextInt();
+@IndexOf("visited") int start=sc.nextInt();
 int visited[]=new int[v];
 
 s.push(start);
 visited[start]=1;
 System.out.print("Path :    "+start+"     ");
-@LTLengthOf("visited") int vertex=1;
+@IndexOf("visited") int vertex=1;
 int cost=0;
 int min=Integer.MAX_VALUE;
 
 while(!s.isEmpty())
 {
 int flag=0;
-@LTLengthOf("mat") int ele=s.peek();
+int ele=s.peek();
```

Comment:
`@LtTengthOf("mat")` was good. But what you wanted to express is that `ele` needs to be an index for `mat`.
Node that `@IndexFor("mat") is interpreted internally by CF as both `@NonNegative` and `@LTLengthOf("mat")`. Hopefully it makes sense.

Reply:
Thanks.I will remember it from next time.